prince-of-versions/src/main/java/co/infinum/princeofversions/PresenterImpl.java

```diff
@@ -0,0 +1,82 @@
+package co.infinum.princeofversions;
+
+import android.support.annotation.VisibleForTesting;
+
+import static co.infinum.princeofversions.UpdateStatus.MANDATORY;
+
+public class PresenterImpl implements Presenter {
+
+    private Interactor interactor;
+
+    private Storage storage;
+
+    public PresenterImpl(Interactor interactor, Storage storage) {
+        this.interactor = interactor;
+        this.storage = storage;
+    }
+
+    @Override
+    public Result check(Loader loader, ApplicationConfiguration appConfig) throws Throwable {
+        return run(loader, appConfig);
+    }
+
+    @Override
+    public PrinceOfVersionsCall check(final Loader loader, Executor executor, final UpdaterCallback callback,
+            final ApplicationConfiguration appConfig) {
+        final PrinceOfVersionsCall call = createCall();
+        executor.execute(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    Result result = PresenterImpl.this.run(loader, appConfig);
+                    if (!call.isCanceled()) {
+                        switch (result.getStatus() != null ? result.getStatus() : UpdateStatus.NO_UPDATE) {
```

Comment:
Is there a reason why we're not just returning the `Result` via a callback here?

Also, maybe we should consider implementing the `PrinceOfVersionsCall` with a similar fashion to the Retrofit `Call`. The call could be executed directly or a callback could be enqueued. Do you think it would be more robust this way?

Reply:
Sounds good! Let's do that